apis/kueue/v1beta1/workload_types.go

```diff
@@ -429,13 +429,13 @@ type WorkloadStatus struct {
 	// +optional
 	ClusterName *string `json:"clusterName,omitempty"`
 
-	// nodesToReplace holds the names of failed nodes running at least one pod of this workload
+	// unhealthyNodes holds the failed nodes running at least one pod of this workload
 	// when Topology-Aware Scheduling is used. This field should not be set by the users.
 	// It indicates Kueue's scheduler is searching for replacements of the failed nodes.
-	// Requires enabling the TASFaliedNodReplacement feature gate.
+	// Requires enabling the TASFailedNodeReplacement feature gate.
 	//
 	// +optional
-	NodesToReplace []string `json:"nodesToReplace,omitempty"`
+	UnhealthyNodes []UnhealthyNode `json:"unhealthyNodes,omitempty"`
```

Comment:
This is just a comment.

We might want to add `//+listType` marker, but we can reconsider which type we should select when we add more fields to this `unhealthyNodes` afterward.

Reply:
Probably we should use //+listType=map for uniqueness by key.

Reply:
`//+listType=map` with Name as the identifier sgtm